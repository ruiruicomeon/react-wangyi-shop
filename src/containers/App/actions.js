import { LOAD_APP } from './constants';

export function loadApp() {
  return {
    type: LOAD_APP,
  };
}
