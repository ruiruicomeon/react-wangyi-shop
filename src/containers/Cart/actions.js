import {
  ADD_TO_CART,
  DEL_FROM_CART,
  CHECK_CART,
  CHANGE_CART_QUANTITY,
  TRANSFER_TO_USER_CART,
  LOAD_CARTS,
  SET_CARTS,
  SET_IS_CHECK_ALL,
  SET_IS_CHECK_ONE,
  CHECK_ALL,
  UN_CHECK_ALL,
} from './constants';

export function addToCart(product) {
  return {
    type: ADD_TO_CART,
    payload: { product },
  };
}

export function changeCartQuantity(product) {
  return {
    type: CHANGE_CART_QUANTITY,
    payload: { product },
  };
}

export function checkCart(product) {
  return {
    type: CHECK_CART,
    payload: { product },
  };
}

export function delFromCart(product) {
  return {
    type: DEL_FROM_CART,
    payload: { product },
  };
}

export function transferToUserCart() {
  return {
    type: TRANSFER_TO_USER_CART,
  };
}

export function loadCarts() {
  return {
    type: LOAD_CARTS,
  };
}

export function setCarts(carts) {
  return {
    type: SET_CARTS,
    payload: { carts },
  };
}

export function setIsCheckAll(isCheckAll) {
  return {
    type: SET_IS_CHECK_ALL,
    payload: { isCheckAll },
  };
}

export function setIsCheckOne(isCheckOne) {
  return {
    type: SET_IS_CHECK_ONE,
    payload: { isCheckOne },
  };
}

export function checkAll() {
  return {
    type: CHECK_ALL,
  };
}

export function unCheckAll() {
  return {
    type: UN_CHECK_ALL,
  };
}
