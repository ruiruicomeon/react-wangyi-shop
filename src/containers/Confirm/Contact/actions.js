import { FETCH_CONTACTS, SET_CONTACTS, SAVE_CONTACT, SET_ERROR, SET_CONTACT, SET_IS_EDIT } from './constants';

export function fetchContacts() {
  return {
    type: FETCH_CONTACTS,
  };
}

export function setContacts(contacts) {
  return {
    type: SET_CONTACTS,
    payload: {
      contacts,
    },
  };
}

export function setError(error) {
  return {
    type: SET_ERROR,
    payload: {
      error,
    },
  };
}

export function saveContact(contact) {
  return {
    type: SAVE_CONTACT,
    payload: {
      contact,
    },
  };
}

export function setContact(contact) {
  return {
    type: SET_CONTACT,
    payload: {
      contact,
    },
  };
}

export function setIsEdit(isEdit) {
  return {
    type: SET_IS_EDIT,
    payload: {
      isEdit,
    },
  };
}
