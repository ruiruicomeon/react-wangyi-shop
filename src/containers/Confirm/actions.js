import { MAKE_ORDER } from './constants';

export function makeOrder() {
  return {
    type: MAKE_ORDER,
  };
}
