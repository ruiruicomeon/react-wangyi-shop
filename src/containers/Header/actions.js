import { FETCH_HOT_WORDS, SET_HOT_WORDS, FETCH_CATEGORIES, SET_CATEGORIES } from './constants';

export function fetchCategories() {
  return {
    type: FETCH_CATEGORIES,
  };
}

export function setCategories(categories) {
  return {
    type: SET_CATEGORIES,
    payload: {
      categories,
    },
  };
}

export function setHotWords(hotWords) {
  return {
    type: SET_HOT_WORDS,
    payload: {
      hotWords,
    },
  };
}

export function fetchHotWords() {
  return {
    type: FETCH_HOT_WORDS,
  };
}
