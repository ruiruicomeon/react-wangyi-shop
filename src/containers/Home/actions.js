import {
  FETCH_CAROUSELS,
  SET_CAROUSELS,
  FETCH_NEW_PRODUCTS,
  SET_NEW_PRODUCTS,
  FETCH_RECOMMEND_PRODUCTS,
  SET_RECOMMEND_PRODUCTS,
  FETCH_BEST_SELL_PRODUCTS,
  SET_BEST_SELL_PRODUCTS,
  FETCH_TIME_PRODUCTS,
  SET_TIME_PRODUCTS,
  FETCH_WELFARE_PRODUCTS,
  SET_WELFARE_PRODUCTS,
  FETCH_PRESENT_PRODUCTS,
  SET_PRESENT_PRODUCTS,
  FETCH_CATEGORY_LIST,
  SET_CATEGORY_LIST,
} from './constants';

export function fetchCarousels() {
  return {
    type: FETCH_CAROUSELS,
  };
}

export function setCarousels(carousels) {
  return {
    type: SET_CAROUSELS,
    payload: {
      carousels,
    },
  };
}

export function fetchNewProducts() {
  return {
    type: FETCH_NEW_PRODUCTS,
  };
}

export function setNewProducts(newProducts) {
  return {
    type: SET_NEW_PRODUCTS,
    payload: {
      newProducts,
    },
  };
}

export function fetchRecommendProducts() {
  return {
    type: FETCH_RECOMMEND_PRODUCTS,
  };
}

export function setRecommendProducts(recommendProducts) {
  return {
    type: SET_RECOMMEND_PRODUCTS,
    payload: {
      recommendProducts,
    },
  };
}

export function fetchBestSellProducts() {
  return {
    type: FETCH_BEST_SELL_PRODUCTS,
  };
}

export function setBestSellProducts(bestSellProducts) {
  return {
    type: SET_BEST_SELL_PRODUCTS,
    payload: {
      bestSellProducts,
    },
  };
}

export function fetchTimeProducts() {
  return {
    type: FETCH_TIME_PRODUCTS,
  };
}

export function setTimeProducts(timeProducts) {
  return {
    type: SET_TIME_PRODUCTS,
    payload: {
      timeProducts,
    },
  };
}

export function fetchWelfareProducts() {
  return {
    type: FETCH_WELFARE_PRODUCTS,
  };
}

export function setWelfareProducts(welfareProducts) {
  return {
    type: SET_WELFARE_PRODUCTS,
    payload: {
      welfareProducts,
    },
  };
}

export function fetchPresentProducts() {
  return {
    type: FETCH_PRESENT_PRODUCTS,
  };
}

export function setPresentProducts(presentProducts) {
  return {
    type: SET_PRESENT_PRODUCTS,
    payload: {
      presentProducts,
    },
  };
}

export function fetchCategoryList() {
  return {
    type: FETCH_CATEGORY_LIST,
  };
}

export function setCategoryList(categoryList) {
  return {
    type: SET_CATEGORY_LIST,
    payload: {
      categoryList,
    },
  };
}
