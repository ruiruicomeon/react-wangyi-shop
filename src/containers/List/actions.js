import { FETCH_CAROUSELS, SET_CAROUSELS, FETCH_PRODUCTS, SET_CATEGORY, SET_THIRD_CATEGORIES, SET_PRODUCTS } from './constants';

export function fetchCarousels(categoryId) {
  return {
    type: FETCH_CAROUSELS,
    payload: {
      categoryId,
    },
  };
}

export function setCarousels(carousels) {
  return {
    type: SET_CAROUSELS,
    payload: {
      carousels,
    },
  };
}

export function fetchProducts(categoryId) {
  return {
    type: FETCH_PRODUCTS,
    payload: {
      categoryId,
    },
  };
}

export function setCategory(category) {
  return {
    type: SET_CATEGORY,
    payload: {
      category,
    },
  };
}

export function setThirdCategories(thirdCategories) {
  return {
    type: SET_THIRD_CATEGORIES,
    payload: {
      thirdCategories,
    },
  };
}

export function setProducts(products) {
  return {
    type: SET_PRODUCTS,
    payload: {
      products,
    },
  };
}
