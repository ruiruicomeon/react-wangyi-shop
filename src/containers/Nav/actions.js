import { FETCH_NOTICES, SET_NOTICES } from './constants';

export function fetchNotices() {
  return {
    type: FETCH_NOTICES,
  };
}

export function setNotices(notices) {
  return {
    type: SET_NOTICES,
    payload: {
      notices,
    },
  };
}
