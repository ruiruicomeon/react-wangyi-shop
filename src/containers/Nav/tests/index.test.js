import React from 'react';
import { Provider } from 'react-redux';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ConnectedRouter } from 'connected-react-router';

import history from '@/utils/history';
import configureStore from '@/store';

import { mapDispatchToProps } from '../index';
import { Nav } from '../index';
import { showLogin, logout } from '@/containers/Login/actions';
import { fetchNotices } from '../actions';

describe('<Nav />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('snapshot', () => {
    const loadSpy = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Nav user={null} notices={null} onLoad={loadSpy} />
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('调用加载函数', () => {
    const loadSpy = jest.fn();
    render(
      <Provider store={store}>
        <Nav onLoad={loadSpy} />
      </Provider>
    );
    expect(loadSpy).toHaveBeenCalled();
  });

  it('点击登陆', () => {
    const loadSpy = jest.fn();
    const loginSpy = jest.fn();
    const { getByText } = render(
      <Provider store={store}>
        <Nav onLoad={loadSpy} onShowLogin={loginSpy} />
      </Provider>
    );
    fireEvent.click(getByText('登录/注册'));

    expect(loginSpy).toHaveBeenCalled();
  });

  describe('dispatch', () => {
    it('调用onShowLogin时dispath showLogin', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onShowLogin();
      expect(dispatch).toHaveBeenCalledWith(showLogin());
    });

    it('调用onLoad时dispath showLogin', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onLoad();
      expect(dispatch).toHaveBeenCalledWith(fetchNotices());
    });

    it('调用onLogout时dispath logout', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onLogout();
      expect(dispatch).toHaveBeenCalledWith(logout());
    });
  });
});
