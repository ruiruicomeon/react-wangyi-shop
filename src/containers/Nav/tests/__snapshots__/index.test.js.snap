// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> snapshot 1`] = `
<nav
  className="nav"
>
  <div
    className="content container"
  >
    <div
      className="notice"
    >
      <img
        alt="speaker"
        className="notice__img"
        src="speaker.gif"
      />
      <ul
        className="notice__list"
      >
        <li
          className="notice__item"
        />
      </ul>
    </div>
    <div
      className="link"
    >
      <ul
        className="link__list"
      >
        <li
          className="link__item"
        >
          <div
            className="link__link"
          >
            登录/注册
          </div>
        </li>
        <li
          className="link__item"
        >
          <div
            className="link__link"
          >
            会员
          </div>
        </li>
        <li
          className="link__item"
        >
          <div
            className="link__link"
          >
            甄选家
          </div>
        </li>
        <li
          className="link__item"
        >
          <div
            className="link__link ant-dropdown-trigger"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            企业采购 
            <span
              aria-label="down"
              className="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </div>
        </li>
        <li
          className="link__item"
        >
          <div
            className="link__link ant-dropdown-trigger"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            客户服务 
            <span
              aria-label="down"
              className="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </div>
        </li>
        <li
          className="link__item"
        >
          <div
            className="link__link"
          >
            <span
              className="link__appIcon"
            />
            APP
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="login"
  />
</nav>
`;
